include/rmm/mr/device/device_memory_resource.hpp

```diff
@@ -306,7 +306,8 @@ class device_memory_resource {
    *
    * @return bool true if the resource supports get_mem_info, false otherwise.
    */
-  [[nodiscard]] virtual bool supports_get_mem_info() const noexcept = 0;
+  //[[deprecated("Use rmm::available_device_memory())")]]  //
```

Comment:
To clarify my understanding, a later PR will uncomment this and deprecate this method?

If so, can we add the deprecation notice in that second PR rather than leaving a commented deprecation notice in this PR? If I came across this code I would think it was _previously_ deprecated and is now un-deprecated by the commenting.

Reply:
@bdice yes, see the plan in https://github.com/rapidsai/rmm/issues/1388 (checklist item 3 is what you're looking for)

I agree with him that we should do this later

Reply:
Related, but possible a comment more relevant to a subsequent PR if we do remove the deprecation notice in this PR: how should we document the difference between querying for the total device memory available and the amount provided by the memory resource? The difficulty in accurately providing the latter is one reason that this API is being removed, but prior users may have expected it to be working. Should we include that explanation somewhere? I think #1388 explains this fairly well, maybe adding it to a PR description (perhaps in the PR actually adding the deprecation) would be sufficient so that the explanation is immortalized in the changelog (i.e. in the git repo rather than only on GH).

Reply:
We document it, yes.  FWIW, at least within RAPIDS, there are no uses of `get_mem_info`.

Reply:
Removed comments. Those were from my initial testing of deprecation. That was intended to surface needed changes in other RAPIDS libraries. "Allowed" deprecation warnings in the other libraries makes this difficult, and a textual search turns out to be easier.